src/main/java/org/opentripplanner/graph_builder/module/interlining/InterlineProcessor.java

```diff
@@ -125,66 +139,122 @@ private Multimap<TripPatternPair, TripPair> getInterlinedTrips(
-            interlines.put(
-              new TripPatternPair(prevPattern, currPattern),
-              new TripPair(prev.getTrip(), curr.getTrip())
-            );
+          if (
+            uncoveredDays.intersects(daysForToTripTimes) &&
+            createInterline(fromTripTimes, toTripTimes, blockId, patternForTripTimes, interlines)
+          ) {
+            uncoveredDays.andNot(daysForToTripTimes);
+            if (uncoveredDays.isEmpty()) {
+              break;
+            }
           }
         }
-        prev = curr;
       }
     }
 
     return interlines;
   }
 
   /**
-   * This compound key object is used when grouping interlining trips together by (serviceId,
-   * blockId).
+   * Validates that trip times are continuous and that the transfer stop(s) are not too far away
+   * from each other. Then creates interline between the trips.
+   *
+   * @return true if interline has been created or if there is an issue preventing an interline
+   * creation for certain service dates.
    */
-  private record BlockIdAndServiceId(String blockId, FeedScopedId serviceId) {
-    static BlockIdAndServiceId ofTrip(Trip trip) {
-      return new BlockIdAndServiceId(trip.getGtfsBlockId(), trip.getServiceId());
+  private boolean createInterline(
+    TripTimes fromTripTimes,
+    TripTimes toTripTimes,
+    String blockId,
+    Map<TripTimes, TripPattern> patternForTripTimes,
+    Multimap<TripPatternPair, TripPair> interlines
+  ) {
+    if (
+      fromTripTimes.getDepartureTime(fromTripTimes.getNumStops() - 1) >
+      toTripTimes.getArrivalTime(0)
+    ) {
+      LOG.error(
+        "Trip times within block {} are not increasing on after trip {}.",
+        blockId,
+        fromTripTimes.getTrip().getId()
+      );
+      return true;
+    }
+    TripPattern fromPattern = patternForTripTimes.get(fromTripTimes);
+    TripPattern toPattern = patternForTripTimes.get(toTripTimes);
+    var fromStop = fromPattern.lastStop();
+    var toStop = toPattern.firstStop();
+    double teleportationDistance = SphericalDistanceLibrary.fastDistance(
+      fromStop.getLat(),
+      fromStop.getLon(),
+      toStop.getLat(),
+      toStop.getLon()
+    );
+    if (teleportationDistance > maxInterlineDistance) {
+      issueStore.add(
+        new InterliningTeleport(
+          fromTripTimes.getTrip(),
+          blockId,
+          (int) teleportationDistance,
+          fromStop,
+          toStop
+        )
+      );
+      // Only skip this particular interline edge; there may be other valid ones in the block for the
+      // from trip.
+      return false;
+    } else {
+      interlines.put(
+        new TripPatternPair(fromPattern, toPattern),
+        new TripPair(fromTripTimes.getTrip(), toTripTimes.getTrip())
+      );
+      return true;
+    }
+  }
+
+  /**
+   * @param mutable This should be set as true if the {@link BitSet} will be modified as otherwise
+   *                we don't have to create a copy of a cached entity.
```

Comment:
This is a bad practice, either create two methods: `getAndCopyDaysForService(FeedScopedId)`and `getDaysForService(FeedScopedId)`, or just do the clone() in the caller. Do not pass in boolean flags to alter method flow.

Reply:
I first had it like that but I thought with one method, it might be more difficult to miss the fact that it should be considered if the BitSet will be modified. I agree that it's bad practice to alter the method flow through a parameter like that so I have implemented your suggestion in https://github.com/opentripplanner/OpenTripPlanner/pull/5162/commits/e40297d407f77e0d8e22f87d036762bdc271c743